src/language-js/parser-babylon.js

```diff
@@ -52,19 +51,59 @@ function createParse(parseMethod, extraPlugins) {
     const combinations = [
       babelOptions(
         { strictMode: true },
-        ["decorators-legacy"].concat(extraPlugins)
+        [
+          "decorators-legacy",
+          ["pipelineOperator", { proposal: "minimal" }]
+        ].concat(extraPlugins)
```

Comment:
This is bad. I added that as an escape-hatch until I could figure out a decent way to detect which decorator plugin to apply.

This is likely cause a noticeable performance degradation, when there's a simple syntax error.

Reply:
@sosukesuzuki Why not always use `proposal: 'smart'`?

Reply:
The `smart` is compatible with `minimal`, but it is coincidental. For example, the same problem will occur again when we try to support `fsarp`.

So, I considered generally solution...

Reply:
As far as I understand, we'll need to check **only** `smart` and `fsharp` then. Why check `minimal`?

Reply:
`smart` is not compatible w/ `minimal`, so that's not a safe assumption. `x |> foo()` is invalid in `smart` but valid in `minimal`.

Reply:
Oh, I see now. That's really unlucky.

Reply:
Oh, I see now. That's really unlucky. Hopefully Babel's error recovery might help here.

If it doesn't help, I'm going to make Prettier try using each proposal until the input parses. To avoid overhead, this logic will be used only if the input contains the substring `|>`.